### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by ExE-Boss**
See https://github.com/sindresorhus/meta/issues/9#issuecomment-491619037

review?(@BendingBender, @sindresorhus)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	| 'rawTrailers'
+	| 'setTimeout'
+	| 'socket'
+	| 'statusCode'
+	| 'statusMessage'
+	| 'trailers'
+	| 'url';
+
+/** Helper type to flatten nested mapped types */
+declare type Identity<T> = {} & {[K in keyof T]: T[K]};
```
Comment (ExE-Boss) [inline index.d.ts:21]: This converts something like: `Pick<IncomingMessage, Exclude<"complete", keyof T> | Exclude<"destroy", keyof T> | Exclude<"headers", keyof T> | Exclude<"httpVersion", keyof T> | Exclude<"httpVersionMinor", keyof T> | Exclude<"httpVersionMajor", keyof T> | Exclude<"method", keyof T> | Exclude<"rawHeaders", keyof T> | Exclude<"rawTrailers", keyof T> | Exclude<"setTimeout", keyof T> | Exclude<"socket", keyof T> | Exclude<"statusCode", keyof T> | Exclude<"statusMessage", keyof T> | Exclude<"trailers", keyof T> | Exclude<"url", keyof T>>` (which is what `mimicResponse` gets interpreted as) into:
```ts
{
    complete: boolean;
    destroy: (error?: Error | undefined) => void;
    headers: http.IncomingHttpHeaders;
    httpVersion: string;
    httpVersionMinor: number;
    httpVersionMajor: number;
    method?: string | undefined;
    ... 7 more ...;
    url?: string | undefined;
}
```

File: index.js [hunk 2]
```diff
@@ -33,4 +33,6 @@ module.exports = (fromStream, toStream) => {
 
 		toStream[prop] = typeof fromStream[prop] === 'function' ? fromStream[prop].bind(fromStream) : fromStream[prop];
 	}
+
+	return toStream;
```
Comment (ExE-Boss) [inline index.js:37]: This makes it&nbsp;possible to&nbsp;alter the&nbsp;type of&nbsp;`toStream`.

**Comment by sindresorhus** [comment]
I appreciate the PR, but I commented to @bendingbender exactly because I prefer him to do it. He knows exactly how to create the type definitions correctly. I don't really have time to review random TS definition submissions.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**